File: 2998-count-symmetric-integers/2998-count-symmetric-integers.py
class Solution:
    def checkSymmetric(self,n):
        if len(str(n)) % 2 != 0 :
            return False
        m = str(n)
        s1 = 0
        for i in range(len(str(n))//2):
            s1 += int(m[i])
        s2 = 0
        for i in range(len(str(n))//2,len(str(n))):
            s2 += int(m[i])
        if s1 == s2 :
            return True
            return False

        
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        c = 0
        s = Solution();
        for i in range(low,high+1):
            if s.checkSymmetric(i) == True:
                print(i)
                c += 1
        return c

        

File: 2998-count-symmetric-integers/2998-count-symmetric-integers.c
int lengthOfNo(int n)
{
    int c = 0;
    while(n)
    {
        c++;
        n /= 10;
    }
    return c;
}

int checkNoOfDigits(int n)
{
    int c = 0;
    while(n)
    {
        c++;
        n /= 10;
    }
    return c%2 == 0;
}
int checkSymmetric(int n)
{   int l = lengthOfNo(n);
    int s1 = 0,s2 = 0;
    int c = 0;
    
    while(n)
    {
        int r = n % 10;
        c++;
        
        if(c <= l/2)
        s1 += r;
        else
        s2 += r;
        n /= 10;
    }
    return s1 == s2;
}
int countSymmetricIntegers(int low, int high) {
    int c=0;
    for(int i=low;i<=high;i++)
    if(checkNoOfDigits(i) && checkSymmetric(i))
    c++;
    return c;

    
}